src/main/java/org/acra/ACRAConfiguration.java

```diff
@@ -320,6 +321,12 @@ public ACRAConfiguration setResDialogNegativeButtonText(int resId) {
         return this;
     }
 
+    @SuppressWarnings( "unused" )
+    public ACRAConfiguration setReportDialogClass(Class<? extends CrashReportDialog> reportDialogClass) {

```

Comment:
Yeah. That's where I have doubts.

I like the idea, don't get me wrong.
But this will open a whole can of maintenance worms.2 out of 3 will
implement it backwards and I'll be fielding the support fall out.
ᐧ

On Fri, Jan 16, 2015 at 9:00 PM, npombourcq <notifications@github.com>
wrote:

> In src/main/java/org/acra/ACRAConfiguration.java
> <https://github.com/ACRA/acra/pull/218#discussion-diff-23074370>:
>
> > @@ -320,6 +321,12 @@ public ACRAConfiguration setResDialogNegativeButtonText(int resId) {
> >          return this;
> >      }
> >
> > +    @SuppressWarnings( "unused" )
> > +    public ACRAConfiguration setReportDialogClass(Class<? extends CrashReportDialog> reportDialogClass) {
>
> At that point it's really up to the custom implementation to decide what
> they want to do. They can extend to get most of the base behavior as is, or
> duplicate from CrashReportDialog if they have somewhat different logic.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ACRA/acra/pull/218/files#r23074370>.
>